app/app.go

```diff
@@ -1011,3 +1012,12 @@ func NewEvmSysContractAddressHandler(ak *evm.Keeper) sdk.EvmSysContractAddressHa
 		return ak.IsMatchSysContractAddress(ctx, addr)
 	}
 }
+
+func NewUpdateCMTxNonceHandler() sdk.UpdateCMTxNonceHandler {
+	return func(tx sdk.Tx, nonce uint64) {
+		stdtx, ok := tx.(*authtypes.StdTx)
+		if ok && nonce != 0 {
```

Comment:
Maybe we should limit the nonce to be positive
```suggestion
		if ok && nonce > 0 {
```

Reply:
it's uint64, so no need >0